docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,62 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
-``context:prefix=target`` arguments, where both the ``context:`` and the
-``=target`` parts are optional (``target`` defaults to ``prefix`` in this
-case). All remapping values that are regular files are compiled (including
-their dependencies).
-
-This mechanism is backwards-compatible (as long
-as no filename contains ``=`` or ``:``) and thus not a breaking change. All
-files in or below the ``context`` directory that import a file that starts with
-``prefix`` are redirected by replacing ``prefix`` by ``target``.
-
-For example, if you clone ``github.com/ethereum/dapp-bin/`` locally to
-``/usr/local/dapp-bin``, you can use the following in your source file:
-
-::
+The compiler recognizes two kinds of imports, based on how the import path looks like:
 
-  import "github.com/ethereum/dapp-bin/library/iterable_mapping.sol" as it_mapping;
+- All the imports mentioned in the previous section are :ref:`direct imports <direct-imports>`.
+  In this case the import path simply becomes the source unit name.
 
-Then run the compiler:
+  Examples:
 
-.. code-block:: bash
+  .. code-block:: solidity
 
-  solc github.com/ethereum/dapp-bin/=/usr/local/dapp-bin/ source.sol
+     import "contracts/lib/token.sol" as token1;
+     import "@contracts/lib/token.sol" as token2;
+     import "/contracts/lib/token.sol" as token3;
+     import "https://example.com/contracts/lib/token.sol" as token4;
 
-As a more complex example, suppose you rely on a module that uses an old
-version of dapp-bin that you checked out to ``/usr/local/dapp-bin_old``, then you can run:
+- The other kind is a :ref:`relative import <relative-imports>`.
+  Such an import must start with ``./`` or ``../``:
 
-.. code-block:: bash
+  .. code-block:: solidity
 
-  solc module1:github.com/ethereum/dapp-bin/=/usr/local/dapp-bin/ \
-       module2:github.com/ethereum/dapp-bin/=/usr/local/dapp-bin_old/ \
-       source.sol
-
-This means that all imports in ``module2`` point to the old version but imports
-in ``module1`` point to the new version.
-
-.. note::
-
-  ``solc`` only allows you to include files from certain directories. They have
-  to be in the directory (or subdirectory) of one of the explicitly specified
-  source files or in the directory (or subdirectory) of a remapping target. If
-  you want to allow direct absolute includes, add the remapping ``/=/``.
-
-If there are multiple remappings that lead to a valid file, the remapping
-with the longest common prefix is chosen.
-
-**Remix**:
-
-`Remix <https://remix.ethereum.org/>`_ provides an automatic remapping for
-GitHub and automatically retrieves the file over the network. You can import
-the iterable mapping as above,  e.g.
-
-::
+     import "./contracts/lib/token.sol" as token1;
+     import "../contracts/lib/token.sol" as token2;
 
-  import "github.com/ethereum/dapp-bin/library/iterable_mapping.sol" as it_mapping;
+  Unlike in case of a direct import, the compiler does assume that the import path is actually a path.
```

Comment:
Isn't this complicating the isuse a little? Whta does it mean to assume it is a path? Isn't the rule just "if it starts with `./` or `../`, append it to the path and simplify"?

Reply:
It assumes it's structured like a path, e.g. expects path separators or squashes `../` segments. It's even processed internally using `boost::filesystem`.